src/cmd/builtin/proxy.c

```diff
@@ -351,13 +359,26 @@ static int cmd_proxy (optparse_t *p, int ac, char *av[])
      */
     save_terminal_state ();
     if (flux_reactor_run (r, 0) < 0) {
+        if (errno == ECONNRESET)
+            log_msg ("Lost connection to Flux");
+        else
+            log_err ("flux_reactor_run");
+        if (optparse_hasopt (p, "kill")) {
+            log_msg ("Sending SIGTERM to child processes");
+            flux_future_destroy (flux_subprocess_kill (ctx.p, SIGTERM));
```

Comment:
I'm not sure of myself here, but would SIGHUP be a better choice than SIGTERM?  Bash ignores SIGTERM but exits on SIGHUP after resending SIGHUP to all jobs.  For an interactive shell, it seems a little nicer to have everything clean up when the connection goes away, instead of being left with a shell with a non-functional FLUX_URI in its `env` (making my nice ƒ(s=4,d=0,builddir) PS1 sad.

Along those lines, what about reversing the sense of the kill option so that signaling is the default, but let it be disabled in cases where it's causing problems?

I realize SIGHUP is supposed to mean "interactive user went away" not "some application that happens to be the center of this developer's universe bu likely not everyone's went away" so it's possible this is a bad idea.

Reply:
Hm, good question. `SIGTERM` seemed to work for me (it causes my interactive shell to immediately exit). I didn't even consider `SIGHUP`, but now that I look at the `bash(1)` man page I see you are right (at least for bash), so I think I agree with your assessment. I'll test it out.

I'm fine with reversing the sense of the `-k, --kill` option. I'll rename it `-k, --no-kill` if that sounds ok?

Reply:
Ah, there's a nice history delivering `SIGHUP` to processes when their "controlling" process exits of which I was only vaguely aware:

https://pubs.opengroup.org/onlinepubs/9699919799/functions/_Exit.html#tag_16_01_08_02

I wonder if we should send SIGHUP+SIGCONT and make believe we're a shell or kernel?

Reply:
Ok, I made the suggested change (and added `SIGCONT` in case the proces is stopped).

The option to avoid `SIGHUP+SIGCONT` is now `-n, --nohup`.

Updated tests and the manpage. I added the following spiel to the manpage:

```

       If the connection to the Flux instance is lost, for  example  when  the
       target instance terminates, flux proxy will emit an error message, send
       SIGHUP and SIGCONT to the spawned shell or other process, and wait  for
       it  to  terminate  before exiting.  The delivery of signals can be dis‐
       abled with the -n, --nohup option, but be aware that Flux commands run‐
       ning  under  a  flux  proxy  which  has lost its connection will likely
       result in errors.
```

Reply:
I like the way `--nohup` turned out there.  Nice.